src/data-version/data-version-middleware.ts

```diff
@@ -1,41 +1,71 @@
 import { DATA_VERSION, PolarisGraphQLContext } from '@enigmatis/polaris-common';
+import { DataVersion, Connection } from '@enigmatis/polaris-typeorm';
+import { PolarisGraphQLLogger } from '@enigmatis/polaris-graphql-logger';
 
-export const dataVersionMiddleware = async (
-    resolve: any,
-    root: any,
-    args: any,
-    context: PolarisGraphQLContext,
-    info: any,
-) => {
-    // if (context.logger) {
-    //     context.logger.debug('Data version middleware started job', { context });
-    // }
-    const result = await resolve(root, args, context, info);
-    let finalResult;
-    if (!root && context.requestHeaders.dataVersion && !isNaN(context.requestHeaders.dataVersion)) {
-        if (Array.isArray(result)) {
-            finalResult = result.filter(entity =>
-                entity.dataVersion && context.requestHeaders.dataVersion
-                    ? entity.dataVersion > context.requestHeaders.dataVersion
-                    : entity,
-            );
-        } else if (
-            !result.dataVersion ||
-            (result.dataVersion &&
+export class DataVersionMiddleware {
+    readonly connection: Connection;
+    readonly logger: PolarisGraphQLLogger;
+
+    constructor(connection: Connection, logger: PolarisGraphQLLogger) {
+        this.connection = connection;
+        this.logger = logger;
+    }
+
+    getMiddleware() {
```

Comment:
Declare return type

Reply:
we need to to npm i @types/apollo-server for this